src/include/homestore/btree/detail/varlen_node.hpp

```diff
@@ -270,15 +270,15 @@ class VariableNode : public BtreeNode {
             vb.bytes = kb.bytes + kb.size;
             vb.size = get_nth_value_len(ind);
 
-            auto sz = other.insert(0, kb, vb); // Keep on inserting on the first index, thus moving everything to right
-            if (!sz) break;
-
-            --ind;
-            ++nmoved;
             if ((kb.size + vb.size + this->get_record_size()) > size_to_move) {
                 // We reached threshold of how much we could move
                 break;
             }
+
```

Comment:
In the current implementation of insert, when there is no enough space for accommodating the new entity, an assert is triggered. To avoid hitting it, the check for available size needs to be performed in advance.

Reply:
<img width="813" alt="Screenshot 2023-10-10 at 10 57 12 AM" src="https://github.com/eBay/HomeStore/assets/116847813/808070c5-9e5c-48c1-9332-c42011382bb1">